### ISSUE TITLE
Comment row at end of query in SQL Lab fails to render in Explore window

### ISSUE THREAD

**Issue by villebro** [apache/incubator-superset#4412]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.23.0dev

### Expected results
When commenting out the last lines in a query in the SQL Lab UI, the query runs just fine. This is rather common when developing new queries to match the exact use case. However, when pressing "Visualize", the query fails due to the "--" leaking into the closing of the subquery. This can be seen in the "View Query" screenshot below. I tried doing a quick hack of adding "\n\n" to the end of the query in the connectors/sqla/models.py file, which fixed the issue (see diff below). I am glad to make a more elegant fix if someone can give me some guidance on the preferred type of fix.

```
diff --git a/superset/connectors/sqla/models.py b/superset/connectors/sqla/models.py
index b2e21c7d..ff1a37a8 100644
--- a/superset/connectors/sqla/models.py
+++ b/superset/connectors/sqla/models.py
@@ -423,7 +423,7 @@ class SqlaTable(Model, BaseDatasource):
     def get_from_clause(self, template_processor=None, db_engine_spec=None):
         # Supporting arbitrary SQL statements in place of tables
         if self.sql:
-            from_sql = self.sql
+            from_sql = self.sql + "\n\n"
             if template_processor:
                 from_sql = template_processor.process_template(from_sql)
             if db_engine_spec:
```
<img width="828" alt="screen shot 2018-02-13 at 9 51 12" src="https://user-images.githubusercontent.com/33317356/36139214-587a7ef0-10a5-11e8-9872-f4c94f5acd84.png">

### Actual results
<img width="830" alt="screen shot 2018-02-13 at 9 50 04" src="https://user-images.githubusercontent.com/33317356/36139202-4fbc88b2-10a5-11e8-895a-d649ecf4c323.png">

<img width="1091" alt="screen shot 2018-02-13 at 11 13 53" src="https://user-images.githubusercontent.com/33317356/36141884-1265492c-10af-11e8-808c-a76f51155c10.png">

### Steps to reproduce
<img width="946" alt="screen shot 2018-02-13 at 9 50 15" src="https://user-images.githubusercontent.com/33317356/36139223-63843d68-10a5-11e8-9979-8636e8cb2545.png">

**Comment by xrmx** [comment]
sqlparse.format has an option for striping comments http://sqlparse.readthedocs.io/en/latest/api/#formatting-of-sql-statements

**Comment by villebro** [comment]
Thanks, that looks much better:

```
diff --git a/superset/connectors/sqla/models.py b/superset/connectors/sqla/models.py
index b2e21c7d..5792498a 100644
--- a/superset/connectors/sqla/models.py
+++ b/superset/connectors/sqla/models.py
@@ -423,7 +423,7 @@ class SqlaTable(Model, BaseDatasource):
     def get_from_clause(self, template_processor=None, db_engine_spec=None):
         # Supporting arbitrary SQL statements in place of tables
         if self.sql:
-            from_sql = self.sql
+            from_sql = sqlparse.format(self.sql, strip_comments=True)
             if template_processor:
                 from_sql = template_processor.process_template(from_sql)
             if db_engine_spec:
```

<img width="1076" alt="screen shot 2018-02-13 at 11 27 39" src="https://user-images.githubusercontent.com/33317356/36142498-f23eb35c-10b0-11e8-8f7e-4d2e6a2f4ad8.png">

Ok for PR?

**Comment by xrmx** [comment]
Looks good to me, bonus points for adding a test :)

**Comment by xrmx** [comment]
Looking again at the code we may want to clean the sql *after*  poking with the template processor.

**Comment by villebro** [comment]
Right you are, I missed that. New try:

```
diff --git a/superset/connectors/sqla/models.py b/superset/connectors/sqla/models.py
index b2e21c7d..9e2ae200 100644
--- a/superset/connectors/sqla/models.py
+++ b/superset/connectors/sqla/models.py
@@ -428,6 +428,7 @@ class SqlaTable(Model, BaseDatasource):
                 from_sql = template_processor.process_template(from_sql)
             if db_engine_spec:
                 from_sql = db_engine_spec.escape_sql(from_sql)
+            from_sql = sqlparse.format(from_sql, strip_comments=True)
             return TextAsFrom(sa.text(from_sql), []).alias('expr_qry')
         return self.get_sqla_table()

```

**Comment by xrmx** [comment]
@villebro please open a PR so it can be reviewed and merged.

**Comment by mistercrunch** [comment]
@villebro fyi is you include `closes #4412` as a commit message in your PR, Github closes the issue(s) as the PR gets merged automatically.
